### PR TITLE
feat(2119): upgrade to helm 3 and add queue-service

### PR DESCRIPTION
**Pull request by tk3fftk**
## Context

We should upgrade helm chart for new possible users.

I only tested my home k8s environment. And I just modified `/etc/hosts` to deploy sdcd with this chart.

## Objective
- Migrate to Helm 3
    - dependencies were moved to `Chart.yaml` from `requirements.yaml`.
    - there are some command change so updating README
- Enabled to deploy queue-service
    - also providing to generate secrets for queue-service
- Remove env (like `prod-`) prefix
    - I think we should not add prefix to resource's name. It should be done by separating namespace. 

## References
- [Helm \| Docs](https://helm.sh/docs/faq/)
- related: https://github.com/screwdriver-cd/screwdriver/issues/2119

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: templates/api.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ .Values.env }}-sdapi
```
Comment (jithine) [inline templates/api.yaml:4]: should we keep env ? What if user wants to setup both prod & beta env ?

Comment (tk3fftk) [reply templates/api.yaml:4]: I think it should be done by separating k8s namespace. 
Force env prefix cause that user have to modify secret YAMLs which are generated by out script.

If we have to keep, how about providing opt-in prefix or helm name prefix?

Comment (jithine) [reply templates/api.yaml:4]: how does that work ? Does helm take care of this with name prefix automatically ?

**Review by jithine — APPROVED** [review]
Good with this now, we can enhance it later on how to support multiple env, like beta & prod.

**Review by pritamstyz4ever — APPROVED** [review]

**Comment by parthasl** [comment]
@tk3fftk 1. secrets will get regenerated when we do multiple environments? 2. how does the table prefix work? Do we use beta- prefix for the beta environment.

**Comment by tk3fftk** [comment]
@jithine @parthasl thank you for your review. I'll take a look about usual chart's prefix.

> secrets will get regenerated when we do multiple environments?

In case of "secrets" mean generated secrets by helm, yes. Because we deploy multiple charts when we want multiple environments.
In case of "secrets" mean screwdriver's secrets, we need manual operation to update YAMLs.

**Comment by tk3fftk** [comment]
@jithine @parthasl @pritamstyz4ever 
I updated the name prefix. It works as`{user-specified-name}-{chart-name}-{resouce-name}` like other charts (e.g., [prometheus](https://github.com/helm/charts/blob/e9d85afa10373b94184e911ebc9eae4e1dfb3bfc/stable/prometheus/templates/_helpers.tpl) and some other charts).

```
~/git/screwdriver-chart(upgrade-for-helm-3*) » k get po -n sd                                                
NAME                                            READY   STATUS      RESTARTS   AGE
3-036j9                                         0/1     Completed   0          9d
3-jdkoh                                         0/1     Completed   0          9d
5-4sd0d                                         0/1     Completed   0          9d
5-ilm5f                                         0/1     Completed   0          9d
7-y3pq8                                         0/1     Completed   0          9d
sd-postgresql-0                                 1/1     Running     0          9d
sd-screwdriver-store-67f9484f55-g4q76           1/1     Running     0          6m21s
sd-screwdriver-ui-75759c7646-n88vg              1/1     Running     0          6m21s
sd-screwdriver-queue-service-747c4478d8-wzv9v   1/1     Running     0          6m21s
sd-screwdriver-api-78dfdf68c6-gdfx7             1/1     Running     0          6m21s
sd-redis-master-0                               1/1     Running     0          6m10s
```

**Comment by tk3fftk** [comment]
I also add screwdriver.yaml. Would you create a pipeline?
it works like this https://cd.screwdriver.cd/pipelines/5098/builds/334948/steps/helm-lint

**Comment by jithine** [comment]
Pipeline created https://cd.screwdriver.cd/pipelines/5099/pulls

**Review by parthasl — APPROVED** [review]
👍

**Comment by jithine** [comment]
@tk3fftk does this address https://github.com/screwdriver-cd/screwdriver/issues/1846 ?

**Comment by tk3fftk** [comment]
@jithine it looks  https://github.com/screwdriver-cd/screwdriver-chart/pull/16 has already fixed https://github.com/screwdriver-cd/screwdriver/issues/1846 😃

**Pull request merged**